src/service.ts

```diff
@@ -0,0 +1,141 @@
+import { DebugSession } from './session';
+
+import { DebugProtocol } from 'vscode-debugprotocol';
+
+import { Debugger } from './debugger';
+
+import { IDebugger } from './tokens';
+
+import { Variables } from './variables';
+import { Callstack } from './callstack';
+
+export class DebugService {
+  constructor(session: DebugSession | null, debuggerModel: Debugger.Model) {
+    this.session = session;
+    this._model = debuggerModel;
+  }
+
+  private _session: DebugSession;
+  private _model: Debugger.Model;
+  frames: any[];
```

Comment:
Should this be `DebugProtocol.StackFrame[]`?

Reply:
It's different its have additional value with converted scope for UI. I will add some interface for that